scripts/checkout.sh

```diff
@@ -19,8 +19,30 @@ checkout() (
 	SRC="$1"
 	REF="$2"
 	REF_FETCH="$REF"
+
+	# git ls-remote's <pattern> argument [1] is a glob [2], and matches anything
+	# ending with the given string. This is problematic if multiple tags or
+	# branches end with the given pattern. In containerd's case, this returns
+	# both tags for the main module ("refs/tags/v1.7.19") and 	# the API module
+	# ("refs/tags/api/v1.7.19").
+	#
+	# To prevent both of those being found, we check if the given reference starts
+	# with a "v"; if it does, we can assume it's a tag, and prefix the pattern with
+	# "refs/tags/" to make it less ambiguous.
+	#
+	# We're using a case statement here to avoid introducing Bashisms.
+	#
+	# [1]: https://git-scm.com/docs/git-ls-remote#Documentation/git-ls-remote.txt-ltpatternsgt82308203
+	# [2]: https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-glob
+	ref_glob="$REF"
+	case $ref_glob in
+		"v"*)
+			ref_glob="refs/tags/$ref_glob"
+			;;
+  esac
```

Comment:
Can't be indented?

Reply:
oh! yes it can; not sure what happened. Let me shfmt it.

Reply:
fixed 👍